src/middlewared/middlewared/plugins/acme_protocol_/authenticators/ovh.py

```diff
@@ -0,0 +1,51 @@
+import logging
+
+from lexicon.providers.ovh import ENDPOINTS
+from certbot_dns_ovh._internal.dns_ovh import _OVHLexiconClient
+
+from middlewared.schema import accepts, Dict, Str, ValidationErrors
+from middlewared.service import skip_arg
+
+from .base import Authenticator
+
+
+logger = logging.getLogger(__name__)
+OVH_ENDPOINTS = tuple(ENDPOINTS.keys())
+
+class OVHAuthenticator(Authenticator):
+
+    NAME = 'OVH'
+    PROPAGATION_DELAY = 60
+    SCHEMA = Dict(
+        'OVH',
+        Str('application_key', empty=False, null=False, title='OVH Application Key'),
+        Str('application_secret', empty=False, null=False, title='OVH Application Secret'),
+        Str('consumer_key', empty=False, null=False, title='OVH Consumer Key'),
+        Str('endpoint', empty=False, default='ovh-eu', title='OVH Endpoint', enum=OVH_ENDPOINTS),
+    )
+
+    def initialize_credentials(self):
+        self.application_key = self.attributes.get('application_key')
+        self.application_secret = self.attributes.get('application_secret')
+        self.consumer_key = self.attributes.get('consumer_key')
+        self.endpoint = self.attributes.get('endpoint')
+
+    @staticmethod
+    @accepts(SCHEMA)
+    @skip_arg(count=1)
+    async def validate_credentials(middleware, data):
+        verrors = ValidationErrors()
+        verrors.check()
+        return data
+
+    def _perform(self, domain, validation_name, validation_content):
+        self.get_client().add_txt_record(domain, validation_name, validation_content)
+
+    def get_client(self):
+        return _OVHLexiconClient(
+            self.endpoint, self.application_key, self.application_secret,
```

Comment:
Looking at this, it seems all these attributes are required - please make the appropriate changes in the schema

Reply:
Sure thing, I did it, it's just that since there wasn't much documentation about what a schema actually is (for truenas) I based myself on the cloudflare authenticator to integrate the OVH authenticator as closely as possible.